### ISSUE TITLE
Define white label instance creation flow and baseline cost

### ISSUE THREAD

**Issue by gusaus** [elmsln/openulmus.org#21]
@btopro provided this overview of Open Ulmus and the course create flow in #drupalopenlearning (Drupal Slack) -  

> Open Ulmus is a deployment of ELMS:LN; providing a series of tools for teaching / training. Collectives would then be able to offer trainings through open ulmus

> Drupal dojo for example might say “we want to offer trainings on drupal” and they use Open ulmus to deliver the training / organize things

>In ELMS:LN land - A new “course” is created which generates a new drupal website that’s of an educationally focused distribution.

>If Dojo says “we want to over a view training and charge 30$” then there would be a mechanism for that (course = drupal site once again here so each separate training = new drupal site that’s stamped out via the ELMS:LN deployment stuff)

>architectural reason for segregated sites is that Training 1 may be completely free (open educational resource) while Trainings 2 and 3 cost 10 and 30 $ respectively; kind of helps limit the scope of each site as far as what it’s responsible for

Assuming this will be clarified and will reside somewhere in the Open Ulmus site, we should include in the project scope https://github.com/elmsln/openulmus.org/issues/1, yes?

**Comment by mmilutinovic013** [comment]
Agreed. I can make the flow on Google Drive tonight and link it in here. You guys are making the docs public/link -- correct? I wasn't sure if there was a folder or not.

**Comment by mmilutinovic013** [comment]
In progress and currently living here: https://drive.google.com/file/d/0Bx3T7Xey416xaVhpbHl3N1lMdUE/view?usp=sharing

**Comment by mmilutinovic013** [comment]
@btopro - can you provide feedback on the overview I have for the flow? I can add more details on how the automation works and any other roles that would touch it. Since ELMS:LN is extremely flexible -- it does not require a lot of people to set up a new course.

**Comment by mmilutinovic013** [comment]
Perhaps screenshots of an install would help? Or a video?

**Comment by btopro** [comment]
@mmilutinovic1313 I don't think there's a need for sections to be used. Rick's people don't really use sections. It's more like a rolling enrolment where you have some system / custom thing that says "Jennie paid, Jennie gets access to this course" sorta deal.

If the course improves, Jennie gets the changes automatically. If the content drastically changes, well then it's not the original training and you'd just make a new space. For example "Drupal 7 Basics" is not the same as "Drupal 8 basics" so there's no reason there should be bleed over between these trainings / courses.

Something custom (CIS customization or whatever) will need to do the "organization wants to make 6 trainings" sorta logic but I mean all the stuff is there to do it. As you said, it's why things have been built the way they are to allow for this kind of flexibility. CIS could be the "open ulmus automated build" in question.

**Comment by mmilutinovic013** [comment]
That's what I was thinking (the CIS part). I will add these concrete examples in tomorrow when I'm working through this more and then send this in Slack and here for feedback from a general crowd.

**Comment by gusaus** [comment]
Continuing with the example described by @btopro https://github.com/elmsln/openulmus.org/issues/21#issue-269765225, if Drupal Dojo wants to offer trainings on Drupal (actually would be Drupal Open Learning), they'd make some sort of request that would generate a new drupal website that’s of an educationally focused distribution. 

Is there a certain monthly cost that Drupal Open Learning would need to pay to spin up and host their instance of Open Ulmus? would there be tiers based on usage? 

If I'm understanding the workflow Drupal Open Learning would be able to offer tiers for courses running on their platform, but how would Open Ulmus recoup the hosting/usage cost for the platform?

**Comment by mmilutinovic013** [comment]
I would propose the following: you charge for hosting either yearly or monthly for usage. It would be at a competitive rate so that the costs are lower to the customers and the business. For example. Take a hosting cost of hosting this on your own (made up number) $50 for hosting on your own. Keep it at $40 and then since you are looking a bunch of customers at this standard rate you are able to exceed the costs of maintaining your system once a few join. I’m on phone so this may not make sense so I’ll clean the comment here later!

**Comment by mmilutinovic013** [comment]
Relates to #24

**Comment by mmilutinovic013** [comment]
https://github.com/btopro/elmsln/tree/master/docs/diagrams

**Comment by mmilutinovic013** [comment]
Using the above diagrams as a baseline for our course creation flows. We can add these to the group folder :)

**Comment by gusaus** [comment]
The following Slack convo between myself and @djfusco touches on the workflows and costs discussed in this issue.

> gusaus [4:31 PM] 
> 
> This obviously isn’t ELMS:LN, but I think the general service model could work for Open Ulmus https://praxis.coop/en/aegirvps-pricing-and-options
> 
> where there’s a base platform with limits on space, ram and stuff. the basic service level is hosting + platform/security updates
> 
> could we do something like that? Base Open Ulmus at $x/month - service would be platform/security updates
> 
> Assuming we could - one way to fund platform development could be to offer early access to our platform via our Open Collective - like prepay for 1yr hosting at a discount
> 
> dfusco [4:49 PM] 
> I don’t see why not. We would need to work out costs as well as “automatic” security updates. That is - we have to decide what impacts happen for certain levels of patches etc
> 
> Managed vs self managed. Some people want to manage it themselves w admin access. Some don’t.
> 
> gusaus [5:03 PM] 
> With these internal projects, we could include even a ballpark estimate for 1yr hosting + security updates, setup, etc. in the project scope. That would buy us some time to fine tune the pricing and service levels.

@btopro @mmilutinovic1313 @cleverington Sound idea? Can we come up with a ballpark estimate that we can include in https://github.com/DrupalOpenLearning/drupalopenlearning/issues/1?

**Comment by gusaus** [comment]
In addition to a white label Open Ulmus for Drupal Open Learning, @kat-wehr needs a course set up for https://github.com/openulmus/openulmus.org/issues/6

As previously discussed https://github.com/openulmus/openulmus.org/issues/21#issuecomment-349784248 could we come up with a ballpark estimate for 1yr hosting + security updates, setup, etc. that we can include in the project scope?

Is this something you could take on @djfusco?

**Comment by gusaus** [comment]
Here's the follow-up convo from Slack...

> dfusco [1:17 PM]
> Saw that earlier. I’ll get back to you
> I need to understand the scope as far as size and scalability. Or maybe that’s what you need.
> What I mean is - how many instances or is this more of a SaaS review?
> 
> gusaus [1:33 PM]
> @dfusco I think what we were talking about previously would be a good model for a SaaS offering. https://github.com/openulmus/openulmus.org/issues/21#issuecomment-349784248
> 
> Think that’s a ways off… right now we want to spin up one course on Open Ulmus and one entire ELMS for Drupal Open Learning. Short term we need to make sure we’re covering the usage cost so @btopro isn’t on the hook for that

> I think this provides another good model for a future service https://teachable.com/pricing

> 
> Don’t think there would be any sort for free plan tho
> 
> But that type of pricing can be a guide for white label instances in the queue https://github.com/openlearninglabs/administration/issues/6#issue-277957596
> 
> dfusco [2:59 PM]
> Okay. Lemme get back to you with some ideas
> 
> gusaus [3:13 PM]
> @dfusco Thanks - I think we should just converse in the issue :point_up_2:, since the main thing we need to do is keep tabs on usage and cover the AWS cost….which I ‘think’ only @btopro has access and is responsible for that

@dfusco Assumuing we're all in agreement that Open Ulmus will not itself provide SaaS https://github.com/openulmus/openulmus.org/issues/58#issuecomment-368741790, we should probably followup in #openlearninglabs

**Comment by gusaus** [comment]
@btopro Another thing we'd like to document here is the process for accessing courses to multiple instances of ELMS:LN. For example, could our "Intro to ELMSN:LN" be accessed from...
https://courses.openulmus.org/elmsln/
https://courses.drupalopenlearning.org/elmsln/ 

Pretty sure there are related discussions scattered about... but if you could provide options here we can get 'em into the guidelines!

**Comment by btopro** [comment]
in current form that's either multiple links pointing to the same site or it's multiple deploys which are leveraging content that's been imported / synced into each via a common git repo (ala HAX desktop app publisher we are working on presently with each ELMS:LN deploy being an end point of sorts to get that content out to people in a consistent environment)

**Comment by gusaus** [comment]
Ha - I think you just answered my followup -

> leveraging content that's been imported / synced into each via a common git repo 

Is that kinda similar to https://github.com/paulhibbitts/demo-grav-course-hub/tree/master/pages/01.home

**Comment by btopro** [comment]
Similar yes. For example
https://github.com/elmsln/documentation

As repo but as site here https://btopro.gitbooks.io/elmsln-documentation/content/

And then imagine it imported into an elmsln space cause we can do that via drush

**Comment by gusaus** [comment]
Nice! So I for the "Intro to ELMS:LN" course we'd 

- [x] Spin up a course for @kat-wehr to do magic on #6 
- [ ] Point it to https://courses.openulmus.org/elmsln/ & https://courses.drupalopenlearning.org/elmsln/

Then in the not too distant future, we can...

- [ ] Set up the course, properly structured (& licensed), in a repo that can be pulled into an ELMS space?

Assuming I'm a bit off in understanding... I think the latter should be how Open Ulmus courses are distributed.

**Comment by btopro** [comment]
sounds reasonable
